packages/plugins/response-cache/src/redis-cache.ts

```diff
@@ -0,0 +1,127 @@
+import Redis from 'ioredis';
+import type { Cache } from './cache';
+
+export type BuildRedisEntityId = (typename: string, id: number | string) => string;
+export type BuildRedisResponseOpsKey = (responseId: string) => string;
+
+export type RedisCacheParameter = {
+  /**
+   * Redis instance
+   * @see Redis.Redis https://github.com/luin/ioredis
+   */
+  redis: Redis.Redis;
+  /**
+   * Maximum amount of items in the cache. Defaults to `Infinity`.
+   */
+  max?: number;
+  /**
+   * Customize how the cache entity id is built.
+   * By default the typename is concatenated with the id e.g. `User:1`
+   */
+  buildRedisEntityId?: BuildRedisEntityId;
+  /**
+   * Customize how the cache key that stores the operations associated with the response is built.
+   * By default `operations` is concatenated with the responseId e.g. `operations:arZm3tCKgGmpu+a5slrpSH9vjSQ=`
+   */
+  buildRedisResponseOpsKey?: BuildRedisResponseOpsKey;
+};
+
+export const createRedisCache = (params: RedisCacheParameter): Cache => {
+  const store = params.redis;
+
+  const buildRedisEntityId = params?.buildRedisEntityId ?? defaultBuildRedisEntityId;
+  const buildRedisResponseOpsKey = params?.buildRedisResponseOpsKey ?? defaultBuildRedisResponseOpsKey;
+
+  function purgeEntity(entity: string): Promise<string[]> {
+    return new Promise(function (resolve, reject) {
+      const keysToPurge: string[] = [entity];
+
+      // find the responseIds for the entity
+      store.smembers(entity).then(function (responseIds) {
+        // and purge each response since they contained the entity data
+        responseIds.forEach(responseId => {
```

Comment:
I chose to push each key, though one could push the array of keys and then flatten at the end ... but since I have to iterate in order to construct the operations key, figured was simpler to push each.